### ISSUE TITLE
Bad compiler warning when code resolves to multiple experimental APIs

### ISSUE THREAD

**Issue by lilyball** [rust-lang/rust#63573]
With Rust 1.36.0, `Range` has two implementations of `is_empty`, one in an inherent impl, the other in an impl for the `std::iter::ExactSizeIterator` trait. Both of these methods are experimental in this version of Rust.

Given the following code:

```rust
fn main() {
    let r = 1u16..10;
    println!("{:?}", r.is_empty());
}
```

This produces the following output:

```
error[E0034]: multiple applicable items in scope
 --> src/main.rs:3:24
  |
3 |     println!("{:?}", r.is_empty());
  |                        ^^^^^^^^ multiple `is_empty` found
  |
  = note: candidate #1 is defined in an impl for the type `std::ops::Range<_>`
  = note: candidate #2 is defined in an impl of the trait `std::iter::ExactSizeIterator` for the type `std::ops::Range<u16>`
  = help: to disambiguate the method call, write `std::iter::ExactSizeIterator::is_empty(r)` instead
```

The problem is this warning does not tell me that either method is experimental. I also get the exact same error with `Range::is_empty`:

```rust
use std::ops::Range;
fn main() {
    let r = 1u16..10;
    println!("{:?}", Range::is_empty(&r));
}
```

Applying the suggested advice to use the `ExactSizeIterator` version produces the expected error (telling me that it's experimental and how to enable it), but the only way to get the expected error when trying to call the inherent impl is to disable the implicit prelude:

```rust
#![no_implicit_prelude]
extern crate std;
fn main() {
    let r = 1u16..10;
    std::println!("{:?}", r.is_empty());
}
```
